### PR TITLE
CHAT-173 Main Nav Guild Integration

### PR DESCRIPTION
**Pull request by Charleswmm**

### CODE REVIEW

File: src/js/components/MainNav/MainNav.js [hunk 1]
```diff
@@ -1,30 +1,95 @@
...
+  const { discordAPIResources, mainNavButtons, mainNavButtonTypes, discordUrls } = config;
+  const { link, addServerButton } = mainNavButtonTypes;
+  const { client, guilds, icons } = discordAPIResources;
+  const { appCDN } = discordUrls;
+
+  const guildData = useDiscordData(guilds, client);
+
+  let addGuildData;
+
+  if (guildData) {
```
Comment (jakxnz) [inline src/js/components/MainNav/MainNav.js:28]: What happens if there isn't guild data?

File: src/js/hooks/useDiscordData.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const { DiscordStore } = useContext(GlobalContext);
+  const [data, setData] = useState(null);
+
+  useEffect(() => {
+    // A flag to make sure state does not change when the component is not mounted
+    let isMounted = true;
+
+    DiscordStore.getData(resource, userType).then((res) => {
+      if (isMounted) {
+        setData(res);
```
Comment (jakxnz) [inline src/js/hooks/useDiscordData.js:16]: This is a pretty opinionated line of code. Hopefully it doesn't limit you too much for future components

File: src/js/components/MainNavItem/MainNavItem.js [hunk 3]
```diff
@@ -22,22 +22,21 @@ export const uiClassNames = {
...
    */
   backgroundImageStyle = () => {
-    const { imageSrc } = this.props;
+    const { icon } = this.props;
 
-    return imageSrc ? {
-      backgroundImage: `url(${imageSrc})`,
+    return icon ? {
+      backgroundImage: `url(${icon})`,
     } : null;
```
Comment (jakxnz) [inline src/js/components/MainNavItem/MainNavItem.js]:
```suggestion
    return icon
        ? { backgroundImage: `url(${icon})` }
        : null;
```

Comment (jakxnz) [reply src/js/components/MainNavItem/MainNavItem.js]: Lining up the ternary operators is much more readable

File: src/js/components/MainNav/MainNav.js [hunk 4]
```diff
@@ -1,30 +1,95 @@
...
+    guildData.sort((a, b) => {
+      const nameA = a.name.toUpperCase();
+      const nameB = b.name.toUpperCase();
+      if (nameA < nameB) {
+        return -1;
+      }
+      if (nameA > nameB) {
+        return 1;
+      }
+      return 0;
```
Comment (jakxnz) [inline src/js/components/MainNav/MainNav.js:39]:
```suggestion
      return nameA < nameB ? 1 : nameA > nameB ? -1 : 0;
```

File: src/js/components/MainNav/MainNav.js [hunk 5]
```diff
@@ -1,30 +1,95 @@
...
+
+    // Set the new buttons in "Config"
+    if (homeButton && addGuildData && utilityButtons) {
+      Config.set({
+        mainNavButtons: [
+          ...homeButton,
+          ...addGuildData,
+          ...utilityButtons,
+        ],
+      });
```
Comment (jakxnz) [inline src/js/components/MainNav/MainNav.js]: So does this still need to go back to config? Or is there a way to do it with cleaner separation of concerns?

File: src/js/components/UserControlPanel/UserControlPanel.js [hunk 6]
```diff
@@ -29,32 +29,6 @@ const UserControlPanel = ({ history }) => {
...
-        },
-      });
-    });
-
-    // The return function is called when the component has unmounted
-    return () => {
-      isMounted = false;
-    };
-  }, [setUserData]);
-
```
Comment (jakxnz) [inline src/js/components/UserControlPanel/UserControlPanel.js:57]: 👋🏻 😁

**Review by jakxnz — APPROVED** [review]
Looking good. It's good enough to merge, I just had some minor comments

File: src/js/components/MainNav/MainNav.js (hunk 1 above)
Comment (Charleswmm) [reply src/js/components/MainNav/MainNav.js:28]: The main nav will render with just the home and utility buttons

File: src/js/components/MainNav/MainNav.js (hunk 4 above)
Comment (Charleswmm) [reply src/js/components/MainNav/MainNav.js:39]: ESLint: Do not nest ternary expressions.(no-nested-ternary)

**Pull request merged**